### PR TITLE
autobrr: add package/module/test for the application autobrr

### PR DESCRIPTION
**Pull request by borgstad**
###### Add [autobrr](https://github.com/autobrr/autobrr) as a module

The application Autobrr has been added as a module. 

Note: The Autobrr project itself does not expose a `package-lock.json`, therefore it has been generated and is version controlled.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by borgstad** [comment]
Closes #224560

**Comment by savyajha** [comment]
@borgstad Brilliant work! Fingers crossed this gets merged soon.

Wrt the generated `package-lock.json`, there is a [pnpm-lock.yaml](https://github.com/autobrr/autobrr/blob/develop/web/pnpm-lock.yaml) in the repo. It might be a good idea to use something like [pnpm-lock-export](https://github.com/NixOS/nixpkgs/tree/master/pkgs/development/web/pnpm-lock-export) for generating a `package-lock.json` during build if possible, I think?

File: nixos/modules/misc/ids.nix [hunk 1]
```diff
@@ -356,7 +356,7 @@ in
       rstudio-server = 324;
       localtimed = 325;
       automatic-timezoned = 326;
-
+      autobrr = 326;
```
Comment (dominikh) [inline nixos/modules/misc/ids.nix:359]: Should this be `327`?

**Comment by SuperSandro2000** [comment]
Closing in favor of #287593

**Pull request closed without merging**